marlowe-playground-client/src/Simulation/BottomPanel.purs

```diff
@@ -0,0 +1,495 @@
+          , text "Static analysis found the following counterexample:"
+          , ul_
+              [ li_
+                  [ spanText "Initial slot: "
+                  , b_ [ spanText (show initialSlot) ]
+                  ]
+              , li_
+                  [ spanText "Offending transaction list: "
+                  , displayTransactionList transactionList
+                  ]
+              , li_
+                  [ spanText "Warnings issued: "
+                  , displayWarningList transactionWarning
+                  ]
+              ]
+          ]
+      Success (R.Error str) ->
+        div [ classes [ ClassName "padded-explanation" ] ]
+          [ h3_ [ text "Error during analysis" ]
+          , text "Analysis failed for the following reason:"
+          , ul_
+              [ li_
+                  [ b_ [ spanText str ]
+                  ]
+              ]
+          ]
+      Failure failure ->
+        div [ classes [ ClassName "padded-explanation" ] ]
+          [ h3_ [ text "Error during analysis" ]
+          , text "Analysis failed for the following reason:"
+          , ul_
+              [ li_
+                  [ b_ [ spanText failure ]
+                  ]
+              ]
+          ]
+      _ -> text "Analysing..."
+
+displayTransactionList :: forall p. String -> HTML p HAction
+displayTransactionList transactionList = case runParser transactionInputList transactionList of
+  Right pTL ->
+    ol_
+      ( do
+          ( TransactionInput
+              { interval: SlotInterval (Slot from) (Slot to)
+            , inputs: inputList
+            }
+          ) <-
+            ((toUnfoldable pTL) :: Array TransactionInput)
+          pure
+            ( li_
+                [ span_
+                    [ b_ [ text "Transaction" ]
+                    , text " with slot interval "
+                    , b_ [ text $ (show from <> " to " <> show to) ]
+                    , if List.null inputList then
+                        text " and no inputs (empty transaction)."
+                      else
+                        text " and inputs:"
+                    ]
+                , if List.null inputList then
+                    text ""
+                  else
+                    displayInputList inputList
+                ]
+            )
+      )
+  Left _ -> code_ [ text transactionList ]
+
+displayInputList :: forall p. List Input -> HTML p HAction
+displayInputList inputList =
+  ol_
+    ( do
+        input <- (toUnfoldable inputList)
+        pure (li_ (displayInput input))
+    )
+
+displayInput :: forall p. Input -> Array (HTML p HAction)
+displayInput (IDeposit (AccountId accNum owner) party tok money) =
+  [ b_ [ text "IDeposit" ]
+  , text " - Party "
+  , b_ [ text $ show party ]
+  , text " deposits "
+  , b_ [ text $ show money ]
+  , text " units of "
+  , b_ [ text $ show tok ]
+  , text " into account "
+  , b_ [ text ((show accNum) <> " of " <> (show owner)) ]
+  , text "."
```

Comment:
This is starting to look sufficiently complex that you might want to consider `purescript-formatting`.

Reply:
I'm trying to imagine what this would look like, it seems like it wouldn't look very good? I would need a type of `forall p i. Format (Array (HTML p i))`

Reply:
The type for formatters would end up being something like...

```
Format (HTML p i) r (Int -> String -> Bool -> r)
```

...for some formatter that expected `Int -> String -> Bool` arguments. I can help you with it/extend the library if you want.

Again, this is a suggested possibility. Up to you.

Reply:
I just feel like I would end up with a load of extra functions for formatting the different types and then have something that looks quite like the existing function but with `<<<` instead of `,`?

Reply:
oh and the extra functions would essentially be versions of the normal `Halogen.HTML` functions so I would have name conflicts, for example `b_`.

Reply:
in this case I'm going to choose not to but I feel disappointed with that decision, at some point I want to use `purescript-formatting`